kolibri/core/content/test/test_content_request.py

```diff
@@ -0,0 +1,192 @@
+            request_data,
+            format="json",
+        )
+        self.assertEqual(ContentDownloadRequest.objects.count(), 1)
+
+        contentnode_id = response_2.data["contentnode_id"]
+        self.assertEqual(contentnode_id, request_data["contentnode_id"])
+
+    def test_create_content_removal_request(self):
+        self.client.login(username="learner", password="password")
+        # Create a download request first
+        # Prepare the request data
+        request_data = {
+            "contentnode_id": "877a1b783fd348bfb87559883e60e9bf",
+            "metadata": {
+                "title": "Sample Title",
+                "file_size": 1024,
+                "learning_activities": ["3dSeJhqs"],
+            },
+        }
+
+        request = self.client.post(reverse("kolibri:core:contentrequest-list"))
+        request.user = self.user
+
+        serializer = ContentDownloadRequestSeralizer(
+            data=request_data, context={"request": request}
+        )
+        serializer.is_valid(raise_exception=True)
+
+        content_download_request = serializer.save()
+        self.assertIsNotNone(content_download_request.id)
+
+        # then, delete it
+        response = self.client.delete(
+            reverse("kolibri:core:contentrequest-list"), pk=content_download_request.id
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertTrue(
+            ContentRemovalRequest.objects.filter(
+                id=content_download_request.id
+            ).exists()
+        )
+
+    def test_no_duplicate_deletion_requests(self):
+        self.client.login(username="learner", password="password")
+
+        # Create a download request first
+        # Prepare the request data
+        request_data = {
+            "contentnode_id": "877a1b783fd348bfb87559883e60e9bf",
+            "metadata": {
+                "title": "Sample Title",
+                "file_size": 1024,
+                "learning_activities": ["3dSeJhqs"],
+            },
+        }
+
+        request = self.client.post(reverse("kolibri:core:contentrequest-list"))
+        request.user = self.user
+
+        serializer = ContentDownloadRequestSeralizer(
+            data=request_data, context={"request": request}
+        )
+        serializer.is_valid(raise_exception=True)
+
+        content_download_request = serializer.save()
+
+        # then, delete it
+        request_id = content_download_request.id
+        response_1 = self.client.delete(
+            reverse("kolibri:core:contentrequest-list"), pk=request_id
+        )
+        response_2 = self.client.delete(
+            reverse("kolibri:core:contentrequest-list"), pk=request_id
+        )
+
+        self.assertEqual(response_1.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(response_2.status_code, status.HTTP_204_NO_CONTENT)
+        self.assertEqual(ContentRemovalRequest.objects.count(), 1)
```

Comment:
Would be good to add a test here that asserts the behaviour above around failed removal requests - this should help be sure that the `exclude` is in the right place.